src/components/Toast/Toast.tsx

```diff
@@ -0,0 +1,47 @@
+import { ReactNode } from "react";
+import { Icon, IconKey } from "src/components/Icon";
+import { Css, Palette } from "src/Css";
+import { useTestIds } from "src/utils";
+import { IconButton } from "../IconButton";
+import { useToastContext } from "./ToastContext";
+
+export type ToastTypes = "error" | "warning" | "success" | "info" | "alert";
+export interface ToastProps {
+  type: ToastTypes;
+  message: ReactNode;
+}
+
+export function Toast(props: ToastProps) {
+  const { message, type } = props;
+  // using toast context to set notice to undefined once the user clicks out of the alert
+  const { setNotice } = useToastContext();
+  const tid = useTestIds(props, "toast");
+
+  return (
+    <div css={{ ...variantStyles[type], ...Css.df.aic.w100.gap2.px2.py2.$ }} {...tid} role="alert">
+      <span css={Css.fs0.plPx(12).$}>
+        <Icon icon={typeToIcon[type]} {...tid.type} color={Palette.Gray900} />
+      </span>
+      <span {...tid.message}>{message}</span>
+      <span css={Css.pr1.add("lineHeight", 0).add("marginLeft", "auto").$}>
```

Comment:
**question:** Curious, what is the need to set `line-height: 0` here?

Reply:
I was using the same pattern as `Snackbar's` `IconButton`  which also has a `line-height: 0`. By removing the `line-height` it makes the X not as vertically centered aligned
![Screen Shot 2022-11-09 at 5 18 10 PM](https://user-images.githubusercontent.com/86630863/200962944-cd010be9-4c22-4bc0-a2ac-8f548ef6de1c.png)

Reply:
Ahh... okay. I remember this issue. Because the icon is an `inline` element, then it tries to inherit some `line-height`, causing the element to be taller than it should. This is a trick to tell it not to do that. I think we could've also made the `span` which contains the `<Icon />` to be `Css.df.aic.$` and that'd achieve the same thing. But either way, was just trying to remember what that was for. Thanks!